tests/test_cli.py

```diff
@@ -266,8 +325,125 @@ def test_cli_collect_filepaths(
+        ),
+        pytest.param(
+            ["--config", os.path.join("config_files", "with_ignore.yml")],
+            True,
+            id="long_config_specifier_w_ignore",
+        ),
+        pytest.param(
+            ["-C", os.path.join("config_files", "without_ignore.yml")],
+            False,
+            id="short_config_specifier_wo_ignore",
+        ),
+        pytest.param(
+            ["--config", os.path.join("config_files", "without_ignore.yml")],
+            False,
+            id="long_config_specifier_wo_ignore",
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    ["ignore_file_flag", "use_ignore_file"],
+    [
+        pytest.param([], False, id="no_ignore_file"),
+        pytest.param(
+            ["-d", os.path.join("config_files", "docstr_ignore.txt")], True, id="short_ignore_file"
+        ),
+        pytest.param(
+            ["--docstr-ignore-file", os.path.join("config_files", "docstr_ignore.txt")],
+            True,
+            id="long_ignore_file",
+        ),
+    ],
+)
+@pytest.mark.usefixtures("cd_tests_dir_fixture")
+@pytest.mark.skipif(
+    sys.version_info < (3, 6), reason="assert_called_once requires python3.6 or later "
+)
+def test_ignore_patterns_files(
+    paths: List[str],
+    config_flag: List[str],
+    use_yml_ignore: Optional[str],
+    ignore_file_flag: List[str],
+    use_ignore_file: Optional[str],
+    runner: CliRunner,
+    mocker,
+):
+    """Test that CLI inputs are correctly interpreted and passed along to
+    :func:`docstr_coverage.cli.collect_filepaths`
+
+    Parameters
+    ----------
+    paths: List[str]
+        Path arguments provided to CLI. These should be made absolute before they are passed to
+        :func:`docstr_coverage.cli.collect_filepaths`
+    config_flag: List[str]
+        CLI option to specify path of yml config file
+    use_yml_ignore: Boolean
+        True iff `config_flag` points to a file with custom ignore patterns
+    ignore_file_flag: List[str]
+        CLI option to specify path of a plain ignore patterns file
+    use_ignore_file: Boolean
+        True iff `ignore_file_flag` points to a file with custom ignore patterns
+    runner: CliRunner
+        Click utility to invoke command line scripts
+    mocker: pytest_mock.MockFixture
+        Mock to check arguments passed to :func:`docstr_coverage.cli.collect_filepaths`"""
+
+    # Check that there is no `.docstr_coverage` file added to the test folder,
+    #   which may be used as default
+    assert not os.path.isfile(".docstr_coverage") and not os.path.isfile(
+        ".docstr.yaml"
+    ), "This test must run in a folder without config or ignore files"
+
+    mock_parse_ig_f = mocker.patch("docstr_coverage.cli.parse_ignore_names_file")
+    parse_ig_from_dict = mocker.patch("docstr_coverage.cli.parse_ignore_patterns_from_dict")
+
+    run_result = runner.invoke(execute, paths + config_flag + ignore_file_flag)
+
+    if use_yml_ignore and use_ignore_file:
+        assert (
```

Comment:
Is the first assertion necessary? It seems like the second and third would be sufficient

Reply:
Well spotted! Thinking purely about *pass/fail* you are absolutely right, the first test is not required.

However, the first assertion allows to print the quite specific error description (thus allowing the easier understanding of the error when debugging). I am a fan of verbose test errors and this one, in particular, was actually useful during my own debugging. No strong feelings though; just tell me if you prefer the less verbose version.

Reply:
Ah, that's an excellent point. Great idea!